api/src/OpenTelemetry/Internal/Logging/Types.hs

```diff
@@ -1,10 +1,15 @@
 {-# LANGUAGE DeriveFunctor #-}
 {-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE NamedFieldPuns #-}
 
 module OpenTelemetry.Internal.Logging.Types (
   LoggerProvider (..),
   Logger (..),
-  LogRecord (..),
+  LogRecord,
```

Comment:
The constructor of `LogRecord` is no longer exported. Any interaction with `LogRecord` must be done through the functions of `ReadableLogRecord` and `ReadWriteLogRecord`.